flink-staging/flink-streaming/flink-storm-compatibility/src/main/java/org/apache/flink/stormcompatibility/api/FlinkSubmitter.java

```diff
@@ -0,0 +1,258 @@
+	 *             if a topology with this name is already running
+	 * @throws InvalidTopologyException
+	 *             if an invalid topology was submitted
+	 */
+	public static void submitTopology(final String name, final Map<?, ?> stormConf, final FlinkTopology topology)
+		throws AlreadyAliveException, InvalidTopologyException {
+		submitTopology(name, stormConf, topology, (SubmitOptions)null, (FlinkProgressListener)null);
+	}
+	
+	/**
+	 * Submits a topology to run on the cluster. A topology runs forever or until explicitly killed.
+	 * 
+	 * @param name
+	 *            the name of the storm.
+	 * @param stormConf
+	 *            the topology-specific configuration. See {@link Config}.
+	 * @param topology
+	 *            the processing to execute.
+	 * @param opts
+	 *            to manipulate the starting of the topology.
+	 * @throws AlreadyAliveException
+	 *             if a topology with this name is already running
+	 * @throws InvalidTopologyException
+	 *             if an invalid topology was submitted
+	 */
+	public static void submitTopology(final String name, final Map<?, ?> stormConf, final FlinkTopology topology, final SubmitOptions opts)
+		throws AlreadyAliveException, InvalidTopologyException {
+		submitTopology(name, stormConf, topology, opts, (FlinkProgressListener)null);
+	}
+	
+	/**
+	 * Submits a topology to run on the cluster. A topology runs forever or until explicitly killed. The given
+	 * {@link FlinkProgressListener} is ignored because progress bars are not supported by Flink.
+	 * 
+	 * 
+	 * @param name
+	 *            the name of the storm.
+	 * @param stormConf
+	 *            the topology-specific configuration. See {@link Config}.
+	 * @param topology
+	 *            the processing to execute.
+	 * @param opts
+	 *            to manipulate the starting of the topology
+	 * @param progressListener
+	 *            to track the progress of the jar upload process
+	 * @throws AlreadyAliveException
+	 *             if a topology with this name is already running
+	 * @throws InvalidTopologyException
+	 *             if an invalid topology was submitted
+	 */
+	@SuppressWarnings({"unchecked", "rawtypes"})
+	public static void submitTopology(final String name, final Map stormConf, final FlinkTopology topology, final SubmitOptions opts, final FlinkProgressListener progressListener)
+		throws AlreadyAliveException, InvalidTopologyException {
+		if(!Utils.isValidConf(stormConf)) {
+			throw new IllegalArgumentException("Storm conf is not valid. Must be json-serializable");
+		}
+		
+		final HashMap localStormConf = new HashMap(stormConf);
+		localStormConf.putAll(Utils.readCommandLineOpts());
+		
+		final Map conf = Utils.readStormConfig();
+		conf.putAll(localStormConf);
+		
+		final String serConf = JSONValue.toJSONString(localStormConf);
+		
+		if(localNimbus != null) {
+			// TODO: check what is the difference about localNimbus vs LocalCluster
+			// LOG.info("Submitting topology " + name + " in local mode");
```

Comment:
I ask at the dev-storm mailing list and got the following answer:

"LocalNimbus is something that is not really used.  It provides a way for the StormSubmitter to be used when submitting a topology to a local mode cluster.  No one uses it and in the past it didn't work, not sure if it works now or not.  It is probably best to just not use it unless you really need to abstract out local mode from the topology code used to submit a topology.
 - Bobby"

We can ignore this case, and I will remove the code.